src/main/java/org/elasticsearch/index/mapper/core/AbstractFieldMapper.java

```diff
@@ -360,6 +368,14 @@ protected String defaultPostingFormat() {
     protected String defaultDocValuesFormat() {
         return null;
     }
+    
+    protected boolean defaultDocValues() {
+        if (writePre2xSettings) {
+            return Defaults.DOC_VALUES;
```

Comment:
Can we change the constant name to make it explicit that it only applies to 1.x indices?